lib/ProofOfPossessionBuilder.ts

```diff
@@ -0,0 +1,26 @@
+import { createProofOfPossession } from './functions';
+import { PROOF_CANT_BE_CONSTRUCTED, ProofOfPossession, ProofOfPossessionOpts } from './types';
+
+export class ProofOfPossessionBuilder {
+  popCallbackOpts?: ProofOfPossessionOpts;
+  popEncoded?: ProofOfPossession;
+
+  withPoPCallbackOpts(popCallbackOpts: ProofOfPossessionOpts): ProofOfPossessionBuilder {
+    this.popCallbackOpts = popCallbackOpts;
+    return this;
+  }
+
+  withPoPEncoded(popEncoded: ProofOfPossession): ProofOfPossessionBuilder {
+    this.popEncoded = popEncoded;
+    return this;
+  }
+
```

Comment:
I am missing the functions to set the kid, iss, clientid in the builder from the IRR, Accesstoken response etc?

Please have a look at the integration code in the wallet. Where the Proof is being created I would expect a builder that can set most options from the existing/available objects, together with some additional info like a kid, client_id value. Then it is up to the actual callback to fetch the private key/did and sign the JWT.

Reply:
the values `iss`, `kid`, `jti`, `iat`, and `exp` are handled in `setJWSDefaults`